base/networking.c

```diff
@@ -601,11 +601,19 @@ port_range_ranges (const char *port_range)
 {
   gchar **split, **point, *range_start, *current;
   array_t *ranges;
-  int tcp;
+  int tcp, err;
 
   if (!port_range)
     return NULL;
 
+  /* port_range needs to be a valid port_range string. */
+  err = validate_port_range (port_range);
+  assert (!err);
```

Comment:
Not sure why you're asserting this.  To me the assertion says that it ret should always be 0, but because this is an exported function we can't guarantee the assertion -- the caller could provide a value that makes the assertion fail.

Reply:
But if the caller provides an invalid port range then things like `"-2,10-8,90000,T:2=2,U"` are valid entries for the function. 
This means, that e.g. for the range "-2" the following unit tests pass:
```
assert_that (range->start, is_equal_to(0));
assert_that (range->end, is_equal_to(2));
```
So the user gets a `range_t` which has 3 ports (0, 1, 2) when supplying the range "-2".

Do you know an alternative for handling such cases?

Reply:
To be clear, I'm saying that the "assert" should be removed, not that the "if (err)" should be removed.